src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -109,6 +110,11 @@ internal sealed class SourceMemberMethodSymbol : SourceMethodSymbol
             _isExpressionBodied = !hasBlockBody && syntax.ExpressionBody != null;
             _refKind = syntax.ReturnType.GetRefKind();
 
+            if (_refKind == RefKind.RefReadOnly)
+            {
+                this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(diagnostics, syntax.ReturnType.Location, modifyCompilation: true);
```

Comment:
MethodChecks looks like a good place.

---
In reply to: [115104899](https://github.com/dotnet/roslyn/pull/18715#discussion_r115104899) [](ancestors = 115104899)